sonic_platform_base/chassis_base.py

```diff
@@ -59,6 +59,27 @@ def get_base_mac(self):
         """
         raise NotImplementedError
 
+    def get_serial_number(self):
+        """
+        Retrieves the hardware serial number for the chassis
+
+        Returns:
+            A string containing the hardware serial number for this chassis.
+        """
+        raise NotImplementedError
+
+    def get_system_eeprom_info(self):
```

Comment:
Do you also plan to add a system_eeprom.py class in this PR?

Reply:
I don't believe we need a system_eeprom.py unless there is a reason.  But we need the same system eeprom functions in module_base.py since a line card or router card most likely has its own SYS EEPROM.  I will add those functions to module_base.py if you're okay with that.

Reply:
I am debating whether the EEPROM should be an object with a base class which we can then create member objects under the objects which contain one (chassis, module, etc.), thus keeping in line with the new object-oriented approach.